apps/backend/src/datasources/stfc/StfcFapDataSource.ts

```diff
@@ -59,18 +84,71 @@ export default class StfcFapDataSource
       )
       .map((user) => user.userId);
 
-    if (ineligibleUserIds.length === 0) {
+    const failedAssignmentUserIds: number[] = [];
+
+    if (ineligibleUserIds.length > 0) {
+      await Promise.all(
+        ineligibleUserIds.map(async (userId) => {
+          return this.stfcUserDataSource
+            .assignSTFCRoleToUser(userId, STFCRolesMap.FAP_REVIEWER)
+            .catch((error) => {
+              logger.logError(
+                'An error occurred while assigning a FAP Member role',
+                error
+              );
+
+              failedAssignmentUserIds.push(userId);
+
+              return null;
+            });
+        })
+      );
+    }
+
+    if (failedAssignmentUserIds.length === 0) {
       return super.assignReviewersToFap(args);
     }
 
-    const ineligibleUsers = await Promise.resolve(
-      this.stfcUserDataSource.getUsersByUserNumbers(ineligibleUserIds)
+    const failedAssignmentUsers = await Promise.resolve(
+      this.stfcUserDataSource.getUsersByUserNumbers(failedAssignmentUserIds)
     );
 
-    const ineligibleUserEmails = ineligibleUsers.map((user) => user.email);
+    const failedAssignmentUserEmails = failedAssignmentUsers.map(
+      (user) => user.email
+    );
 
     throw new GraphQLError(
-      `Some members: ${ineligibleUserEmails} did not have the correct Roles`
+      `Some members: ${failedAssignmentUserEmails} were unable to be assigned the correct role`
     );
   }
+
+  stfcRoleMap = new Map<UserRole, STFCRolesMap>([
+    [UserRole.FAP_CHAIR, STFCRolesMap.FAP_CHAIR],
+    [UserRole.FAP_SECRETARY, STFCRolesMap.FAP_SECRETARY],
+    [UserRole.FAP_REVIEWER, STFCRolesMap.FAP_REVIEWER],
+  ]);
+
+  async removeMemberFromFap(args: UpdateMemberFapArgs): Promise<Fap> {
+    const role = UserRoleShortCodeMap[args.roleId];
+    const faps = await this.getUserFaps(args.memberId, role);
+
```

Comment:
Could you add a comment here explaining that if the user's last position in a FAP is removed, then the accompanying role is also revoked?